### PR TITLE
Convert remote files

### LINKED ISSUE #72: Can't extract one image from remote file
```
        $url = 'http://www.axmag.com/download/pdfurl-guide.pdf';
        $pdf = new Pdf($url);
        var_dump($pdf->getNumberOfPages());
        $pdf->saveImage('1.jpg');
```
Looks like it adds `[0]` to the URL, because number of pages is working. Is it possible to workaround it? Or I need to download the file manually?

### PR DESCRIPTION
**Pull request by introwit**
- Fixes #72

### CODE REVIEW

File: src/Pdf.php [hunk 1]
```diff
@@ -218,6 +222,27 @@ public function getImageData($pathToImage)
     }
 
     /**
+     * Return raw image data.
+     * @param string $pathToImage
+     *
+     * @return \Imagick
+     */
+    public function convertRemoteFile($pathToImage)
```
Comment (freekmurze) [inline src/Pdf.php]: We don't want users to call this directly. Make this protected.

There also no conversion being done here. Name this `getRemoteImageData`, reflecting the function named where it is being called.

File: src/Pdf.php [hunk 2]
```diff
@@ -218,6 +222,27 @@ public function getImageData($pathToImage)
     }
 
     /**
+     * Return raw image data.
+     * @param string $pathToImage
```
Comment (freekmurze) [inline src/Pdf.php]: Missing newline.

File: src/Pdf.php [hunk 3]
```diff
@@ -218,6 +222,27 @@ public function getImageData($pathToImage)
     }
 
     /**
+     * Return raw image data.
```
Comment (freekmurze) [inline src/Pdf.php]: Let's be more specific, change this to `Return remote raw image data.`

**Comment by introwit** [comment]
@freekmurze done, seems like installing imagick fails, known thing? I guess it's php*V*-imagick

**Comment by freekmurze** [comment]
@introwit yeah, if you know a good solution for those failing tests, send a PR.

**Pull request merged**